scripts/benchmark_plot.py

```diff
@@ -0,0 +1,179 @@
+    errors = np.zeros([len(backends), 2, n_unique])
+    for i in range(0, len(backends)):
+        for j in range(0, len(all_num_primitives[i])):
+            index = num_unique_primitives.index(all_num_primitives[i][j])
+            rates[i, index] = all_rates[i][j]/y_scaling
+            errors[i, 0, index] = 1 - all_errors[i][j]/all_rates[i][j]
+            errors[i, 1, index] = 1 + all_errors[i][j]/all_rates[i][j]
+
+    # Create TeX-style labels
+    num_primitives_labels = [None] * n_unique
+    for i in range(n_unique):
+        num_primitives_labels[i] = '$10^' + str(round(math.log(num_unique_primitives[i], 10))) + '$'
+
+    # Phase 2: plot generation
+    ax = plt.subplot(111)
+    patterns = ('/', 'x', '\\', '.', 'o', 'O')
+
+    def autolabel(rects):
+        """Attach a text label above each bar in *rects*, displaying its height."""
+        for rect in rects:
+            height = round(rect.get_height(),1)
+            ax.annotate('{}'.format(height),
+                        xy=(rect.get_x() + rect.get_width() / 2, rect.get_height()),
+                        xytext=(0, 3),  # 3 points vertical offset
+                        textcoords="offset points",
+                        ha='center', va='bottom',
+                        fontsize=18)
+
+    width = 0.6
+    scale = 1/len(backends)
+    for i in range(0, len(backends)):
+        barplot = ax.bar(np.arange(n_unique) + scale*(i-.5*(len(backends)-1)) * width,
+                         width=scale*width*np.ones(n_unique), height=rates[i, :],
+                         yerr=(errors[i, :] if plot_errors == True else None),
+                         capsize=5, hatch=patterns[i % len(patterns)],
+                         edgecolor='k',
+                         error_kw=dict(ecolor='gray', lw=2, capsize=5, capthick=2),
+                         label=backends[i])
+
+        if plot_numbers:
+            autolabel(barplot)
+
+    ax.legend(loc='upper left', fontsize=20)
+
+    ax.set_xlabel('Number of objects indexed', fontsize=20)
+    plt.xticks(fontsize=18)
+    ax.set_xticks(np.arange(n_unique))
+    ax.set_xticklabels(num_primitives_labels)
+    ax.tick_params(axis='x', which='major', bottom='off', top='off')
+    ax.set_xlim([-0.5, n_unique - 0.5])
+
+    ax.set_ylabel('Rate (million objects/second)', fontsize=20)
+    plt.yticks(fontsize=18)
+    ax.yaxis.grid('on')
+    ax.set_ylim([0, 1.1*ax.get_ylim()[1]])
+
+if __name__ == '__main__':
+
+    # Process input
+    options = docopt(__doc__)
+
+    algorithm = options['--algo']
+    backends = options['--backends']
+    geometry = options['--geometry']
+    input_files = options['--input-files']
+    output_file = options['--output-file']
+    plot_numbers = options['--numbers']
+    plot_errors = options['--errors']
+
+    plt.figure(figsize=(6, 5))
+
+    # Check input
+    if geometry not in ['filled_box', 'hollow_box']:
+        raise Exception('Unknown geometry: "' + geometry + '"')
```

Comment:
```suggestion
        raise Exception('Unknown geometry: "' + geometry + '". Allowed values are "filled_box" and "hollow_box".')
```

Reply:
Allowed values are seen in `./benchmark_plot.py --help`, no need for them here.

Reply:
I know. Just my user experience: If I misspell an option so I prefer knowing what the correct spelling/allowed options are immediately without needing to relaunch with `--help`.